### PR TITLE
Wa corrections for demo

### PR DESCRIPTION
**Pull request by repovic48**

### CODE REVIEW

File: project-generated-code-backend/Backend/Model/Accounts/Account.cs [hunk 1]
```diff
@@ -17,13 +17,13 @@ public abstract class Account : Entity
...
+        public String fullName;
         [ForeignKey("Address")] public string AddressSerialNumber { get; set; }
         public virtual Address Address { get; set; }
         public String Password { get; set; }
-        
-        // this property does not need to be kept in databse
-        public string FullName => Name + " " + Surname;
 
+        // this property does not need to be kept in databse
+        public string FullName { get => fullName; set => fullName=value; }
```
Comment (milicasiriski) [inline project-generated-code-backend/Backend/Model/Accounts/Account.cs]: I can't let this go to develop, the full name field should not exist

File: project-generated-code-backend/Backend/Model/Accounts/Patient.cs [hunk 2]
```diff
@@ -34,7 +34,7 @@ public class Patient : Account
         public bool Guest { get; set; }
         public bool EmailConfirmed { get; set; }
         [ForeignKey("Physician")] public string PhysicianSerialNumber { get; set; }
-        public Physician ChosenPhysician { get; set; }
+        public String ChosenPhysician { get; set; }
```
Comment (milicasiriski) [inline project-generated-code-backend/Backend/Model/Accounts/Patient.cs:37]: You can add a string property chosen physician like:
public string ChosenPhysicianName => $"{ChosenPhysician.Name} {ChosenPhysician.Surname}"
but without a setter, we need to keep the model clean

File: project-generated-code-backend/Backend/Model/Accounts/Physician.cs [hunk 3]
```diff
@@ -52,6 +60,14 @@ public Physician(string name, string surname, string id)
             Specialization = specialization ?? new List<Specialization>();
         }
 
+        public Physician(string name, string surname, Specialization specialization)
+        {
+            this.Name = name;
+            this.Surname = surname;
```
Comment (milicasiriski) [inline project-generated-code-backend/Backend/Model/Accounts/Physician.cs]: call base(name, surname) instead

File: project-generated-code-backend/Backend/Model/Schedule/Appointment.cs [hunk 4]
```diff
@@ -122,9 +138,9 @@ public override string ToString()
         {
             return "patient: " + Patient.FullName + "\nphysitian: " + Physician.FullName + "\ntime interval: " +
                    TimeInterval + "\nroom: " + Room + "\nprocedure type: " + ProcedureType;
-        }
+            }
 
-        public int CompareTo(Appointment other)
+            public int CompareTo(Appointment other)
```
Comment (milicasiriski) [inline project-generated-code-backend/Backend/Model/Schedule/Appointment.cs]: Formatting is off here

File: project-generated-code-backend/Backend/Repository/DatabaseSql/AppointmentDatabaseSql.cs [hunk 5]
```diff
@@ -10,8 +10,8 @@
 
 namespace HealthClinicBackend.Backend.Repository.DatabaseSql
 {
-    public class AppointmentDatabaseSql : GenericDatabaseSql<Appointment>, IAppointmentRepository
-    {
+    public class AppointmentDatabaseSql : GenericDatabaseSql<Appointment> //IAppointmentRepository
```
Comment (milicasiriski) [inline project-generated-code-backend/Backend/Repository/DatabaseSql/AppointmentDatabaseSql.cs]: What happened here?

File: project-generated-code-backend/Backend/Model/Schedule/Appointment.cs [hunk 6]
```diff
@@ -30,14 +30,30 @@ public class Appointment : Entity
         public bool Active { get; set; }
         public string DateOfCanceling { get; set; }
 
+        public bool IsSurveyDone { get; set; }
+        
         public Appointment(Room room, Physician physician, Patient patient, TimeInterval timeInterval,
-            ProcedureType procedureType) : base()
+            ProcedureType procedureType, Boolean active, DateTime date) : base()
```
Comment (milicasiriski) [inline project-generated-code-backend/Backend/Model/Schedule/Appointment.cs:36]: Please create a new constructor that contains active and date, then you won't need to comment the backend code 😄

**Review by milicasiriski — CHANGES_REQUESTED** [review]
The new changes seem ok, but what is going on with appointment services and repository? I don't see you are using it anywhere but it's all in the comments, was there some problem with that? We can go on a call for that if necessary

File: project-generated-code-backend/Backend/Model/Schedule/Appointment.cs (hunk 6 above)
Comment (repovic48) [reply project-generated-code-backend/Backend/Model/Schedule/Appointment.cs:36]: We need additional explanation, we do not understand what the problem is.

File: project-generated-code-backend/Backend/Model/Accounts/Patient.cs (hunk 2 above)
Comment (repovic48) [reply project-generated-code-backend/Backend/Model/Accounts/Patient.cs:37]: We don't understand what we need to do.

File: project-generated-code-backend/Backend/Model/Schedule/Appointment.cs (hunk 6 above)
Comment (milicasiriski) [reply project-generated-code-backend/Backend/Model/Schedule/Appointment.cs:36]: I thought this was the reason for commenting all those services, but if it's not causing any problems then it's fine

File: project-generated-code-backend/Backend/Repository/DatabaseSql/HealthCareSystemDbContext.cs [hunk 7]
```diff
@@ -111,7 +111,7 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
             modelBuilder.Entity<Physician>().HasOne(p => p.Address).WithMany();
             modelBuilder.Entity<Patient>().HasOne(p => p.Address).WithMany();
             modelBuilder.Entity<Secretary>().HasOne(s => s.Address).WithMany();
-
+            /*
```
Comment (milicasiriski) [inline project-generated-code-backend/Backend/Repository/DatabaseSql/HealthCareSystemDbContext.cs]: this

File: project-generated-code-backend/Backend/Repository/FileSystem/AppointmentFileSystem.cs [hunk 8]
```diff
@@ -24,39 +24,39 @@ public List<Appointment> GetAppointmentsByDate(DateTime date)
...
             List<Appointment> appointmentsByDate = new List<Appointment>();
             List<Appointment> allAppointments = GetAll();
-            foreach (Appointment a in allAppointments)
-            {
-                if (date.Equals(a.TimeInterval.Start.Date))
-                {
-                    appointmentsByDate.Add(a);
-                }
-            }
+            //foreach (Appointment a in allAppointments)
```
Comment (milicasiriski) [inline project-generated-code-backend/Backend/Repository/FileSystem/AppointmentFileSystem.cs]: this

File: project-generated-code-backend/Backend/Repository/FileSystem/AppointmentFileSystem.cs [hunk 9]
```diff
@@ -24,39 +24,39 @@ public List<Appointment> GetAppointmentsByDate(DateTime date)
...
+            //{
```
Comment (milicasiriski) [inline project-generated-code-backend/Backend/Repository/FileSystem/AppointmentFileSystem.cs]: this

File: project-generated-code-backend/Backend/Repository/FileSystem/AppointmentFileSystem.cs [hunk 10]
```diff
@@ -24,39 +24,39 @@ public List<Appointment> GetAppointmentsByDate(DateTime date)
...
+            //    if (date.Equals(a.TimeInterval.Start.Date))
```
Comment (milicasiriski) [inline project-generated-code-backend/Backend/Repository/FileSystem/AppointmentFileSystem.cs]: this

File: project-generated-code-backend/Backend/Repository/FileSystem/AppointmentFileSystem.cs [hunk 11]
```diff
@@ -24,39 +24,39 @@ public List<Appointment> GetAppointmentsByDate(DateTime date)
...
+            //    {
```
Comment (milicasiriski) [inline project-generated-code-backend/Backend/Repository/FileSystem/AppointmentFileSystem.cs]: this

File: project-generated-code-backend/Backend/Service/HospitalAccountsService/PatientAccountsService.cs [hunk 12]
```diff
@@ -42,10 +42,10 @@ private bool IsPatientScheduledForPhysitian(Patient patient, Physician physician
             List<Appointment> patientAppointments = _appointmentRepository.GetAppointmentsByPatient(patient);
             foreach (Appointment appointment in patientAppointments)
             {
-                if (appointment.Physician.Equals(physician))
-                {
-                    return true;
-                }
+                //if (appointment.Physician.Equals(physician))
```
Comment (milicasiriski) [inline project-generated-code-backend/Backend/Service/HospitalAccountsService/PatientAccountsService.cs]: this

File: project-generated-code-backend/Backend/Service/SchedulingService/AppointmentGeneralitiesOptions/PatientAvailabilityService.cs [hunk 13]
```diff
@@ -14,7 +14,7 @@ class PatientAvailabilityService
 
         public PatientAvailabilityService()
         {
-            _appointmentRepository = new AppointmentDatabaseSql();
+           // _appointmentRepository = new AppointmentDatabaseSql();
```
Comment (milicasiriski) [inline project-generated-code-backend/Backend/Service/SchedulingService/AppointmentGeneralitiesOptions/PatientAvailabilityService.cs]: this

File: project-generated-code-backend/Backend/Service/SchedulingService/AppointmentGeneralitiesOptions/PhysicianAvailabilityService.cs [hunk 14]
```diff
@@ -13,7 +13,7 @@ public class PhysicianAvailabilityService
 
         public PhysicianAvailabilityService()
         {
-            _appointmentRepository = new AppointmentDatabaseSql();
+           // _appointmentRepository = new AppointmentDatabaseSql();
```
Comment (milicasiriski) [inline project-generated-code-backend/Backend/Service/SchedulingService/AppointmentGeneralitiesOptions/PhysicianAvailabilityService.cs]: this

File: project-generated-code-backend/Backend/Service/SchedulingService/AppointmentGeneralitiesOptions/PhysicianAvailabilityService.cs [hunk 15]
```diff
@@ -57,10 +57,10 @@ private bool IsPhysicianScheduled(Physician physician, TimeInterval timeInterval
             List<Appointment> appointments = _appointmentRepository.GetAppointmentsByPhysician(physician);
             foreach (Appointment appointment in appointments)
             {
-                if (timeInterval.IsOverLapping(appointment.TimeInterval))
-                {
-                    return true;
-                }
+                //if (timeInterval.IsOverLapping(appointment.TimeInterval))
```
Comment (milicasiriski) [inline project-generated-code-backend/Backend/Service/SchedulingService/AppointmentGeneralitiesOptions/PhysicianAvailabilityService.cs]: this

File: project-generated-code-backend/Backend/Service/SchedulingService/AppointmentGeneralitiesOptions/RoomAvailabilityService.cs [hunk 16]
```diff
@@ -17,7 +17,7 @@ public class RoomAvailabilityService
 
         public RoomAvailabilityService()
         {
-            _appointmentRepository = new AppointmentDatabaseSql(); 
+           // _appointmentRepository = new AppointmentDatabaseSql();
```
Comment (milicasiriski) [inline project-generated-code-backend/Backend/Service/SchedulingService/AppointmentGeneralitiesOptions/RoomAvailabilityService.cs]: this

File: project-generated-code-backend/Backend/Service/SchedulingService/AppointmentService.cs [hunk 17]
```diff
@@ -18,7 +18,7 @@ public class AppointmentService
 
         public AppointmentService()
         {
-            _appointmentRepository = new AppointmentDatabaseSql();
+          //  _appointmentRepository = new AppointmentDatabaseSql();
```
Comment (milicasiriski) [inline project-generated-code-backend/Backend/Service/SchedulingService/AppointmentService.cs]: this

File: project-generated-code-backend/Backend/Service/SchedulingService/PatientScheduleService.cs [hunk 18]
```diff
@@ -17,7 +17,7 @@ class PatientScheduleService
         public PatientScheduleService(Patient loggedPatient)
         {
             _loggedPatient = loggedPatient;
-            _appointmentRepository = new AppointmentDatabaseSql();
+          //  _appointmentRepository = new AppointmentDatabaseSql();
```
Comment (milicasiriski) [inline project-generated-code-backend/Backend/Service/SchedulingService/PatientScheduleService.cs]: this

File: project-generated-code-backend/Backend/Service/SchedulingService/PhysicianScheduleService.cs [hunk 19]
```diff
@@ -23,7 +23,7 @@ public class PhysicianScheduleService
         public PhysicianScheduleService(Physician loggedPhysician)
         {
             _loggedPhysician = loggedPhysician;
-            _appointmentRepository = new AppointmentDatabaseSql();
+            //_appointmentRepository = new AppointmentDatabaseSql();
```
Comment (milicasiriski) [inline project-generated-code-backend/Backend/Service/SchedulingService/PhysicianScheduleService.cs]: this

**Review by milicasiriski — CHANGES_REQUESTED** [review]

**Review by milicasiriski — APPROVED** [review]

**Review by markovidovic — APPROVED** [review]

**Review by lukaDoric — APPROVED** [review]

**Pull request merged**